server/utils/util.go

```diff
@@ -0,0 +1,119 @@
+package utils
+
+import (
+	"crypto/rand"
+	"encoding/hex"
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"strings"
+)
+
+var l = log.New(os.Stderr, "", 1)
+
+// Logs error to Stderr.
+func LogError(a ...interface{}) {
+	l.Println(a...)
+}
+
+// Generates Random string using crypto/rand.
+func CryptoRandomString(len int) (string, error) {
+	random := make([]byte, len)
+	_, err := rand.Reader.Read(random)
+	if err != nil {
+		return "", err
+	}
+	return hex.EncodeToString(random), nil
+}
+
+// Enables content security policy which only allows self scripts as well as specific nonces for inline scripts.
+func EnableContentSecurityPolicy(w *http.ResponseWriter) {
+	ContentSecurityPolicyString := strings.Join([]string{
+		"script-src 'self' 'nonce-EDNnf03nceIOfn39fn3e9h3sdfa' 'nonce-afds3h9e3nf93nf0Iecn30fnNDE'",
+		"style-src 'self' 'unsafe-inline' fonts.googleapis.com ",
+		"object-src 'none'"}, ";")
+	(*w).Header().Set("Content-Security-Policy", ContentSecurityPolicyString)
+}
+
+// Gets header from request based on key.
+func GetHeader(r *http.Request, key string) string {
+	return r.Header.Get(key)
+}
+
+// Extracts a token from a cookie based on token name.
+func ExtractTokenFromCookie(tokenName string, r *http.Request) (string, error) {
+	cookie, err := r.Cookie(tokenName)
+	if err != nil {
+			return "", errors.New(tokenName + " not found")
+	}
+	return cookie.Value, nil
+}
+
+// Saves a token cookie with httpOnly and Secure flag.
+func SaveCookie(cookieName, token string, w http.ResponseWriter) {
+	cookie := &http.Cookie{
+		Name:     cookieName,
+		Value:    token,
+		MaxAge:   0,
+		SameSite: http.SameSiteNoneMode,
+		HttpOnly: true,
+		Secure:   true,
+		Path:     "/",
+	}
+	http.SetCookie(w, cookie)
+	w.Header().Add("Vary", "Cookie")
+}
+
+// Gets body from request.
+func GetBody(r *http.Request) ([]byte, error) {
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+			return nil, err
+	}
+	return body, nil
+}
+
+// Serves Json Response.
+func ServeJson(w http.ResponseWriter, obj interface{}) {
+	data, err := json.Marshal(obj)
+	if err != nil {
+			LogError(err)
+			return
+	}
+	_, err = w.Write(data)
+	if err != nil {
+			LogError(err)
+			return
+	}
+}
+
+func ServeCreated(w http.ResponseWriter) {
+	w.WriteHeader(http.StatusCreated)
```

Comment:
nit: also write something in the response body, here and below
you can use `http.Error(w, http.StatusText(http.StatusCreated), http.StatusCreated)`

Reply:
are you suggesting that I return a response body in case of error and possibly log /or other it on the front end?

Reply:
yes, if this response ends up rendered by the browser, the user might see a blank page and not see the status code.
a minimal response body would make the error status more prominent.

and sorry, I highlighted the wrong function. `ServeCreated` is fine without a body, but `ServeUnauthorized` (and other error responses) could use a body

Reply:
Got it, Will fix this.